File: datasets/php/HumanEval_39_prime_fib.php
<?php
// prime_fib returns n-th number that is a Fibonacci number and it's also prime.
// >>> prime_fib(1)
// 2
// >>> prime_fib(2)
// 3
// >>> prime_fib(3)
// 5
// >>> prime_fib(4)
// 13
// >>> prime_fib(5)
// 89
function prime_fib($n) {
    $fib = [1, 2];
    $i = 2;
    while ($i < $n) {
        $fib[] = $fib[$i - 1] + $fib[$i - 2];
        $i++;
    }
    $fib = array_filter($fib, function ($x) {
        return is_prime($x);
    });
    return $fib[$n - 1];
}

function candidate(...$args) {
    return prime_fib(...$args);
}

function test(): void {
    if (candidate(1) !== 2) { throw new Exception("Test failed!"); }
    if (candidate(2) !== 3) { throw new Exception("Test failed!"); }
    if (candidate(3) !== 5) { throw new Exception("Test failed!"); }
    if (candidate(4) !== 13) { throw new Exception("Test failed!"); }
    if (candidate(5) !== 89) { throw new Exception("Test failed!"); }
    if (candidate(6) !== 233) { throw new Exception("Test failed!"); }
    if (candidate(7) !== 1597) { throw new Exception("Test failed!"); }
    if (candidate(8) !== 28657) { throw new Exception("Test failed!"); }
    if (candidate(9) !== 514229) { throw new Exception("Test failed!"); }
    if (candidate(10) !== 433494437) { throw new Exception("Test failed!"); }
}

test();

File: datasets/java/HumanEval_39_prime_fib.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
class Problem {
private static <T> T getDefaultValue(Class<T> clazz) {
return (T) Array.get(Array.newInstance(clazz, 1), 0);
}    // prime_fib returns n-th number that is a Fibonacci number and it's also prime.
    // >>> prime_fib(1)
    // 2
    // >>> prime_fib(2)
    // 3
    // >>> prime_fib(3)
    // 5
    // >>> prime_fib(4)
    // 13
    // >>> prime_fib(5)
    // 89
    public static long prime_fib(long n) {
        if (n <= 0) {
            return getDefaultValue(Long.class);
        }
        if (n == 1) {
            return 2;
        }
        if (n == 2) {
            return 3;
        }
        long fib = 2;
        long prev = 1;
        long temp = 0;
        for (int i = 3; i <= n; i++) {
            temp = fib;
            fib += prev;
            prev = temp;
        }
        return fib;
    }
    // >>> is_prime(2)
    // True
    // >>> is_prime(3)
    // True
    // >>> is_prime(4)
    // False
    // >>> is_prime(5)
    // True
    // >>> is_prime(6)
    // False
    // >>> is_prime(7)
    // True
    // >>> is_prime(8)
    // False
    // >>> is_prime(9)
    // False
    // >>> is_prime(10)
    // False
    // >>> is_prime(11)
    // True
    public static boolean is_prime(long n) {
        if (n <= 1) {
            return getDefaultValue(Boolean.class);
        }
        if (n == 2) {
            return true;
        }
        if (n % 2 == 0) {
            return false;
        }
        for (long i = 3; i * i <= n; i += 2) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
    // >>> is_prime_fib(1)
    // False
    // >>> is_prime_fib(2)
    // True
    // >>> is_prime_fib(3)
    // False
    // >>> is_prime_fib(4)
    // False
    // >>> is_prime_fib(5)
    // True
    // >>> is_prime_fib(6)
    // False
    // >>> is_prime_fib(7

    }
    public static void main(String[] args) {
    assert(prime_fib((1l)) == (2l));
    assert(prime_fib((2l)) == (3l));
    assert(prime_fib((3l)) == (5l));
    assert(prime_fib((4l)) == (13l));
    assert(prime_fib((5l)) == (89l));
    assert(prime_fib((6l)) == (233l));
    assert(prime_fib((7l)) == (1597l));
    assert(prime_fib((8l)) == (28657l));
    assert(prime_fib((9l)) == (514229l));
    assert(prime_fib((10l)) == (433494437l));
    }

}
